ext/odbc/php_odbc.c

```diff
@@ -1810,6 +1810,9 @@ static void php_odbc_fetch_hash(INTERNAL_FUNCTION_PARAMETERS, int result_type)
 
 				if (rc == SQL_SUCCESS_WITH_INFO) {
 					ZVAL_STRINGL(&tmp, buf, result->longreadlen);
+				} else if (rc != SQL_SUCCESS) {
+					php_error_docref(NULL, E_WARNING, "Can't get data of column #%d (retcode %u)", i + 1, rc);
```

Comment:
I can't suggest any objectively better message :) But I have a few notes:
- I think the word `cannot` has a clear dominance over `can't`
- Personally, I prefer `retrieve` (and `fetch`) to `get`: it's probably just me, but the former ones sounds more "professional"
- is the `data` needed in the message?

So I could image a few other variations: `Cannot retrieve/fetch data of column ...`, `Cannot retrieve/fetch column ...` Feel free to disregard these if you are happier with the original one. :)

Reply:
Thanks for the suggestions! Actually, I started with "can't fetch column ...", but changed that to stick more closely to `SQLGetData()` which is the failing function in this case. I think it's a reasonable compromize to hint at that function, short of mentioning the function name directly (what would be too low-level a detail). I agree that "cannot" is an improvement.